src/installer/corehost/corehost.cpp

```diff
@@ -213,11 +213,24 @@ int exe_start(const int argc, const pal::char_t* argv[])
         trace::info(_X("App path: [%s]"), app_path.c_str());
 
         hostfxr_set_error_writer_fn set_error_writer_fn = reinterpret_cast<hostfxr_set_error_writer_fn>(pal::get_symbol(fxr, "hostfxr_set_error_writer"));
-
         {
             propagate_error_writer_t propagate_error_writer_to_hostfxr(set_error_writer_fn);
 
             rc = main_fn_v2(argc, argv, host_path_cstr, dotnet_root_cstr, app_path_cstr);
+
+            if (rc == StatusCode::FrameworkMissingFailure && !set_error_writer_fn)
+            {
+                pal::string_t missing_fw = L"Microsoft.NETCore.App";
+                pal::string_t fw_ver = _STRINGIFY(CUREXE_PKG_VER);
```

Comment:
I think it is possible for the apphost version to be different from the framework version? If so, this could be a bit of a lie and unfortunately, we don't know the exact framework version at this point, so I'm not sure what could be a better alternative.

We currently have two dialog message scenarios:
- missing framework (specific framework not found)
- missing runtime (no hostfxr found at all)

We could:
- go through the missing framework path (what is in this change now)
  - version listed might not be correct (I think?)
- go through the missing runtime path
  - may be a bit confusing, since the message just says .NET Core needs to be installed and the user already has some version of it installed
- add different message for this case (hostfxr < 3.0, FrameworkMissingFailure)
  - not sure what exactly this should convey. Maybe that a compatible version of .NET Core was not found?

Reply:
I definitely see the potential confusion. I think we could change the error message a little bit to note that the version is only approximate. I think the apphost version should pretty much always be equal or lower to the required runtime version (if the app was built with our SDK), so if we word it as "approximate" it should be good enough.